v2/internal/reconcilers/arm/azure_generic_arm_reconciler_instance.go

```diff
@@ -175,7 +176,9 @@ func (r *azureDeploymentReconcilerInstance) DetermineDeleteAction() (DeleteActio
 	return DeleteActionBeginDelete, r.StartDeleteOfResource, nil
 }
 
-func (r *azureDeploymentReconcilerInstance) DetermineCreateOrUpdateAction() (CreateOrUpdateAction, CreateOrUpdateActionFunc, error) {
+func (r *azureDeploymentReconcilerInstance) DetermineCreateOrUpdateAction(
+	ctx context.Context,
```

Comment:
minor: After rejig, you can now remove this `ctx` as this method doesn't make any external calls so has no need for `ctx` (param is unused).